src/bp/ui-shared-lite/Payloads/index.ts

```diff
@@ -0,0 +1,162 @@
+import * as sdk from 'botpress/sdk'
+import { omit } from 'lodash'
+
+const isBpUrl = (str: string): boolean => {
+  const re = /^\/api\/.*\/bots\/.*\/media\/.*/g
+
+  return re.test(str)
+}
+
+// Duplicate of modules/builtin/src/content-types/_utils.js
+const formatUrl = (baseUrl: string, url: string): string => {
+  if (isBpUrl(url)) {
+    return `${baseUrl}${url}`
+  } else {
+    return url
+  }
+}
+
+export const renderPayload = payload => {
+  const type = payload?.type
+
+  if (type === 'single-choice' && payload.choices) {
+    return renderChoicePayload(payload)
+  } else if (type === 'dropdown') {
+    return renderDropdownPayload(payload)
+  } else if (type === 'image' && payload.image) {
+    return renderImagePayload(payload)
+  } else if (type === 'audio' && payload.audio) {
+    return renderAudioPayload(payload)
+  } else if (type === 'video' && payload.video) {
+    return renderVideoPayload(payload)
+  } else if (type === 'file' && payload.file) {
+    return renderFilePayload(payload)
+  } else if (type === 'card') {
+    return renderCarouselPayload({ ...payload, items: [payload] })
+  } else if (type === 'carousel' && payload.items) {
+    return renderCarouselPayload(payload)
+  }
+
+  return payload
+}
+
+const renderChoicePayload = (content: sdk.ChoiceContent) => {
+  if ((content as any).isDropdown) {
+    return {
+      type: 'custom',
+      module: 'extensions',
+      component: 'Dropdown',
+      message: content.text,
+      buttonText: '',
+      displayInKeyboard: true,
+      options: content.choices.map(c => ({ label: c.title, value: c.value.toUpperCase() })),
+      width: 300,
+      placeholderText: (content as any).dropdownPlaceholder
+    }
+  }
+  return {
+    type: 'custom',
+    module: 'channel-web',
+    component: 'QuickReplies',
+    quick_replies: content.choices.map(c => ({
+      title: c.title,
+      payload: c.value.toUpperCase()
+    })),
+    disableFreeText: (content as any).disableFreeText,
+    wrapped: {
+      type: 'text',
+      ...omit(content, 'choices', 'type')
+    }
+  }
+}
+
+const renderDropdownPayload = (content: any) => {
+  // TODO: add typings for dropdowns
+  return {
+    type: 'custom',
+    module: 'extensions',
+    component: 'Dropdown',
+    message: content.message,
+    buttonText: content.buttonText,
+    displayInKeyboard: content.displayInKeyboard,
+    options: content.options,
+    allowCreation: content.allowCreation,
+    allowMultiple: content.allowMultiple,
+    width: content.width,
+    collectFeedback: content.collectFeedback,
+    placeholderText: content.placeholderText,
+    markdown: content.markdown
```

Comment:
Please use deconstructor... we don't really care if other properties are passed down to a frontend component..

```js
return {
    type: 'custom',
    module: 'extensions',
    component: 'Dropdown',
   ...content
}
```